chore: Add PR template, semantic PR config file, update contribution guidelines (#4585) (#4755)
keptn/keptn: CONTRIBUTING.md

For contributions to Keptn, please fork the Keptn repository and work in a branch. More information about forking is written down in the [docs/fork](docs/fork.md).

We are loosely following the [Git Flow branching model](https://nvie.com/posts/a-successful-git-branching-model/) however, we try to avoid the extra step of the `develop` branch and instead work directly on the `master` branch.

> * If you work on a new feature, [fork the repository](docs/fork.md), branch away from the `master` branch and use the following schema for naming your branches:

Reviewer: ```suggestion
* If you work on an improvement, [fork the repository](docs/fork.md), branch away from the `master` branch and use the following schema for naming your branches:
```

Author: done

git push --set-upstream origin feature/123/foo
```

Finally, go to GitHub and create a Pull Request. There should be a PR template already prepared for you.
If not, you will find it at `.github/pull_request_template.md`.
Please describe what this PR is about and add a link to relevant GitHub issues.
If you changed something that is visible to the user, please add a screenshot.
Please follow the [conventional commit guidelines](https://www.conventionalcommits.org/en/v1.0.0/) for your PR title.

If you only have one commit in your PR, please follow the guidelines for the message of that single commit, otherwise the PR title is enough.
You can find a list of all possible feature types [here](#commit-types-and-scopes).

An example for a pull request title would be:
```
feat(api): New endpoint for feature X (#1234)
```
This would be a PR that adds a new endpoint to the keptn API and the issue number related to this PR is #1234.

If you have **breaking changes** in your PR, it is important to note them in the PR description but also in the merge commit for that PR.

> When pressing "squash and merge", you have the option to fill out the commit message. Please use that feature to add the breaking changes according to the conventional commit guidelines.

Reviewer: ```suggestion
When pressing "squash and merge", you have the option to fill out the commit message. Please use that feature to add the breaking changes according to the [conventional commit guidelines](-Please add the link here-).
```

Author: done
